tarogarden/caretaker.go

```diff
@@ -323,10 +323,25 @@ func (b *BatchCaretaker) seedlingsToAssetSprouts(ctx context.Context,
 		}
 
 		var groupKey *asset.GroupKey
+		// If the seedling has a group key specified and emission
+		// enabled, that group key was validated earlier. We need to
+		// sign the new genesis with that group key.
+		if seedling.HasGroupKey() {
+			groupKey, err = asset.DeriveGroupKey(
```

Comment:
hm, feels weird that we must "derive" the group key if we already have it. Could we fetch it instead? :)

```go
groupKey = seedling.GroupInfo.GroupKey.GroupKeyPub
```

Reply:
The naming is a bit weird, `DeriveGroupKey` is the exposed method to actually generate the signature and tweaked key for the asset issuance.

At this point, we have both geneses and the raw key info in `seedling.GroupInfo`, as well as the original signature for the group (which we don't need). We could technically copy the RawKey and GroupPubKey over directly but we have to call `DeriveGroupKey` to sign the new genesis anyways.